build-logic/convention/src/main/java/AndroidApplicationConventionPlugin.kt

```diff
@@ -0,0 +1,38 @@
+import com.android.build.api.dsl.ApplicationExtension
+import com.nibalk.tasky.buildlogic.convention.ExtensionType
+import com.nibalk.tasky.buildlogic.convention.configureBuildTypes
+import com.nibalk.tasky.buildlogic.convention.configureKotlinAndroid
+import com.nibalk.tasky.buildlogic.convention.libs
+import org.gradle.api.Plugin
+import org.gradle.api.Project
+import org.gradle.kotlin.dsl.configure
+
+class AndroidApplicationConventionPlugin: Plugin<Project> {
+    override fun apply(target: Project) {
+        target.run {
+            pluginManager.run {
+                apply("com.android.application")
+                apply("org.jetbrains.kotlin.android")
+                // apply(libs.findPlugin("android.application").get().toString())
+                // apply(libs.findPlugin("jetbrains.kotlin.android").get().toString())
```

Comment:
While this could work, you don't really gain anything out of referring to a plugin like this because plugins are resolved differently than libraries. When adding a library reference in a Gradle file you need to also declare its version which you get with `libs.findLibary(..)`, but when adding a plugin, the version is already contained on the classpath by adding the plugin to your project level Gradle file (but not applying it yet).
So either way, you'll need the plugin ID here in the Gradle file.